hookman/hookman_generator.py

```diff
@@ -711,6 +712,31 @@ def _generate_windows_body(hooks):
         f"    }}",
         f"",
         f"",
+        f"    HMODULE Load_DLL(const std::wstring& w_filename) {{",
+        f'        wchar_t *path_env = NULL;',
+        f'        rsize_t path_env_len = 0;',
+        f'        // Read PATH environment variable',
+        f'        _wdupenv_s(&path_env, &path_env_len, L"PATH");',
+        f'        std::wstring::size_type dir_name_size = w_filename.find_last_of(L"/\\\\");',
+        f'        int size = path_env_len + MAX_PATH;',
+        f'        wchar_t *new_path_env;',
+        f'        new_path_env = (wchar_t *)malloc(size);',
```

Comment:
I think it is probably better to do all string manipulations here using a `std::wstring`; this avoids having to manage memory and use those `wcsncpy*` functions

Reply:
Done.